app/scenes/more_dms/more_dms.js

```diff
@@ -0,0 +1,196 @@
+import {
+    InteractionManager,
+    StyleSheet,
+    TouchableOpacity,
+    View
+} from 'react-native';
+
+import FormattedText from 'app/components/formatted_text';
+import MemberList from 'app/components/member_list';
+import SearchBar from 'app/components/search_bar';
+
+import {Constants, RequestStatus} from 'service/constants';
+import {changeOpacity} from 'app/utils/theme';
+
+const style = StyleSheet.create({
+    container: {
+        flex: 1,
+        backgroundColor: '#fff'
+    }
+});
+
+class MoreDirectMessages extends PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        preferences: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        profiles: PropTypes.array,
+        search: PropTypes.array,
+        requestStatus: PropTypes.object.isRequired,
+        searchRequest: PropTypes.object.isRequired,
+        subscribeToHeaderEvent: React.PropTypes.func.isRequired,
+        unsubscribeFromHeaderEvent: React.PropTypes.func.isRequired,
+        actions: PropTypes.shape({
+            goBack: PropTypes.func.isRequired,
+            makeDirectChannel: PropTypes.func.isRequired,
+            getProfiles: PropTypes.func.isRequired,
+            searchProfiles: PropTypes.func.isRequired
+        }).isRequired
+    };
+
+    static navigationProps = {
+        renderLeftComponent: (props, emitter, theme) => {
+            return (
+                <TouchableOpacity
+                    style={{flex: 1, paddingHorizontal: 15, justifyContent: 'center'}}
+                    onPress={() => emitter('close')}
+                >
+                    <FormattedText
+                        id='more_direct_channels.close'
+                        defaultMessage='Close'
+                        style={{color: theme.sidebarHeaderTextColor}}
+                    />
+                </TouchableOpacity>
+            );
+        }
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.searchTimeoutId = 0;
+
+        this.state = {
+            profiles: [],
+            page: 0,
+            next: true,
+            searching: false
+        };
+    }
+
+    componentWillMount() {
+        this.props.subscribeToHeaderEvent('close', this.props.actions.goBack);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const {requestStatus} = this.props;
+        if (requestStatus.status === RequestStatus.STARTED &&
+            nextProps.requestStatus.status === RequestStatus.SUCCESS) {
+            const {page} = this.state;
+            const profiles = nextProps.profiles.splice(0, (page + 1) * Constants.PROFILE_CHUNK_SIZE);
+            this.setState({profiles});
+        } else if (this.state.searching &&
+            nextProps.searchRequest.status === RequestStatus.SUCCESS) {
+            this.setState({profiles: nextProps.search});
+        }
+    }
+
+    componentDidMount() {
+        InteractionManager.runAfterInteractions(() => {
```

Comment:
cause if we don't do that the modal will freeze the transition until we get the profiles